src/app/state/importaccounts/index.ts

```diff
@@ -15,23 +23,37 @@ const slice = createSlice({
       state.step = undefined
       state.showAccountsSelectionModal = false
     },
-    enumerateAccountsFromLedger(state, action: PayloadAction<void>) {
+    enumerateFirstAccountsFromLedger(state, action: PayloadAction<void>) {
       state.step = undefined
       state.accounts = []
       state.showAccountsSelectionModal = true
     },
+    enumerateMoreAccountsFromLedger(state, action: PayloadAction<void>) {
+      state.step = undefined
+    },
     enumerateAccountsFromMnemonic(state, action: PayloadAction<string>) {
       state.step = undefined
       state.accounts = []
+      state.accountsSelectionPageNumber = 0
       state.showAccountsSelectionModal = true
     },
-    toggleAccount(state, action: PayloadAction<number>) {
-      const index = action.payload
-      state.accounts[index].selected = !state.accounts[index].selected
+    toggleAccount(state, action: PayloadAction<string>) {
+      const account = state.accounts.find(a => a.address === action.payload)!
+      account.selected = !account.selected
+    },
+    accountGenerated(state, action: PayloadAction<ImportAccountsListAccount>) {
+      state.accounts.push(action.payload)
     },
     accountsListed(state, action: PayloadAction<ImportAccountsListAccount[]>) {
       state.accounts = action.payload
     },
+    updateAccount(state, action: PayloadAction<ImportAccountChange>) {
+      const change = action.payload
+      Object.assign(state.accounts[change.index], change)
```

Comment:
we care only about balance 
`state.accounts[action.payload.index].balance = action.payload.balance`

Reply:
Yes, thanks, already fixed that.